org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/firefox/FirefoxLaunchableAdapterFactory.java

```diff
@@ -0,0 +1,28 @@
+package org.eclipse.wildwebdeveloper.debug.firefox;
+
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.Adapters;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.eclipse.debug.ui.actions.ILaunchable;
+
+public class FirefoxLaunchableAdapterFactory implements IAdapterFactory {
+	
+	private static final ILaunchable DUMMY = new ILaunchable() {
+	};
+
+	@Override
+	public <T> T getAdapter(Object adaptableObject, Class<T> adapterType) {
+		IResource resource = Adapters.adapt(adaptableObject, IResource.class);
+		if (adapterType.equals(ILaunchable.class) && Tester.isHTMLProject(resource.getProject())) {
```

Comment:
We also need to return some ILaunchable for HTML files.

Reply:
So I should check `IsLaunchableHTMLTester.isHTMLFile(resource)` and return the ILaunchable?

Reply:
Sounds good. Let's try and see ;)